### PR TITLE
Add documentation about hosting gRPC in non-ASP.NET Core projects

### PR DESCRIPTION
**Pull request by JamesNK**
Fixes https://github.com/grpc/grpc-dotnet/issues/2147

*** EDIT by @Rick-Anderson****
[Internal review URL](https://review.learn.microsoft.com/en-us/aspnet/core/grpc/aspnetcore?view=aspnetcore-8.0&branch=pr-en-us-30390&tabs=visual-studio#host-grpc-in-non-aspnet-core-projects)

---

#### Internal previews

| 📄 File | 🔗 Preview link |
|:--|:--|
| [aspnetcore/grpc/aspnetcore.md](https://github.com/dotnet/AspNetCore.Docs/blob/67cea786da26b8ce43e2fa4a121485760d8223d3/aspnetcore/grpc/aspnetcore.md) | [gRPC services with ASP.NET Core](https://review.learn.microsoft.com/en-us/aspnet/core/grpc/aspnetcore?branch=pr-en-us-30390) |
| [aspnetcore/grpc/interprocess.md](https://github.com/dotnet/AspNetCore.Docs/blob/67cea786da26b8ce43e2fa4a121485760d8223d3/aspnetcore/grpc/interprocess.md) | [Inter-process communication with gRPC](https://review.learn.microsoft.com/en-us/aspnet/core/grpc/interprocess?branch=pr-en-us-30390) |

### CODE REVIEW

File: aspnetcore/grpc/aspnetcore.md [hunk 1]
```diff
@@ -246,6 +246,25 @@ For more information on enabling HTTP/2 and TLS with Kestrel, see [Kestrel endpo
 
 :::moniker-end
 
+## Host gRPC in non-ASP.NET Core projects
+
+An ASP.NET Core gRPC server is usually created from the gRPC template. The project file created by the template uses `Microsoft.NET.SDK.Web` as the SDK:
```
Comment (Rick-Anderson) [inline aspnetcore/grpc/aspnetcore.md:251]:
```suggestion
An ASP.NET Core gRPC server is typically created from the gRPC template. The project file created by the template uses `Microsoft.NET.SDK.Web` as the SDK:
```

File: aspnetcore/grpc/aspnetcore.md [hunk 2]
```diff
@@ -246,6 +246,25 @@ For more information on enabling HTTP/2 and TLS with Kestrel, see [Kestrel endpo
...
+
+[!code-xml[](~/grpc/aspnetcore/Server-web.csproj?highlight=1)]
+
+The `Microsoft.NET.SDK.Web` SDK value automatically adds a reference to the ASP.NET Core framework. The reference allows the app to use ASP.NET Core types required to host a server.
+
+It's also possible to add a server to existing non-ASP.NET Core projects. Add the `Microsoft.AspNetCore.App` framework to the project file:
```
Comment (Rick-Anderson) [inline aspnetcore/grpc/aspnetcore.md:257]:
```suggestion
You can add a gRPC server to non-ASP.NET Core projects with the following project file settings:
```

File: aspnetcore/grpc/aspnetcore.md [hunk 3]
```diff
@@ -246,6 +246,25 @@ For more information on enabling HTTP/2 and TLS with Kestrel, see [Kestrel endpo
...
+
+[!code-xml[](~/grpc/aspnetcore/Server.csproj?highlight=7)]
```
Comment (Rick-Anderson) [inline aspnetcore/grpc/aspnetcore.md:259]:
```suggestion
[!code-xml[](~/grpc/aspnetcore/Server.csproj?highlight=1,7)]
```

Comment (Rick-Anderson) [reply aspnetcore/grpc/aspnetcore.md:259]: Highlight change from above

**Review by Rick-Anderson — APPROVED** [review]
See #30399. We might want to copy that information to the dotnet docs repo.

File: aspnetcore/grpc/aspnetcore.md [hunk 4]
```diff
@@ -246,6 +246,25 @@ For more information on enabling HTTP/2 and TLS with Kestrel, see [Kestrel endpo
...
+
+The `Microsoft.NET.SDK.Web` SDK value automatically adds a reference to the ASP.NET Core framework. The reference allows the app to use ASP.NET Core types required to host a server.
+
+You can add a gRPC server to non-ASP.NET Core projects with the following project file settings:
+
+[!code-xml[](~/grpc/aspnetcore/Server.csproj?highlight=1,7)]
+
+The preceding project file:
+
+* Adds a framework reference to `Microsoft.AspNetCore.App`. The framework reference allows non-ASP.NET Core apps, such as Windows Services, WPF apps, or WinForms apps to use ASP.NET Core APIs. The app can now use ASP.NET Core APIs to start an ASP.NET Core server.
```
Comment (JamesNK) [inline aspnetcore/grpc/aspnetcore.md:263]:
```suggestion
* Adds a framework reference to `Microsoft.AspNetCore.App`. The framework reference allows non-ASP.NET Core apps, such as Windows Services, WPF apps, or WinForms apps to use ASP.NET Core APIs. The app can now use ASP.NET Core APIs to start an ASP.NET Core server. For more information about using the `Microsoft.AspNetCore.App` framework reference, see [Use the ASP.NET Core shared framework
](xref:fundamentals/target-aspnetcore#use-the-aspnet-core-shared-framework).
```

File: aspnetcore/grpc/aspnetcore.md [hunk 5]
```diff
@@ -246,6 +246,26 @@ For more information on enabling HTTP/2 and TLS with Kestrel, see [Kestrel endpo
...
+The `Microsoft.NET.SDK.Web` SDK value automatically adds a reference to the ASP.NET Core framework. The reference allows the app to use ASP.NET Core types required to host a server.
+
+You can add a gRPC server to non-ASP.NET Core projects with the following project file settings:
+
+[!code-xml[](~/grpc/aspnetcore/Server.csproj?highlight=1,7)]
+
+The preceding project file:
+
+* Adds a framework reference to `Microsoft.AspNetCore.App`. The framework reference allows non-ASP.NET Core apps, such as Windows Services, WPF apps, or WinForms apps to use ASP.NET Core APIs. The app can now use ASP.NET Core APIs to start an ASP.NET Core server. For more information about using the `Microsoft.AspNetCore.App` framework reference, see [Use the ASP.NET Core shared framework
+](xref:fundamentals/target-aspnetcore#use-the-aspnet-core-shared-framework).
```
Comment (JamesNK) [inline aspnetcore/grpc/aspnetcore.md:264]:
```suggestion
* Adds a framework reference to `Microsoft.AspNetCore.App`. For more information, see [Use the ASP.NET Core shared framework](xref:fundamentals/target-aspnetcore#use-the-aspnet-core-shared-framework).
  * The framework reference allows non-ASP.NET Core apps, such as Windows Services, WPF apps, or WinForms apps to use ASP.NET Core APIs.
  * The app can now use ASP.NET Core APIs to start an ASP.NET Core server.
```

File: aspnetcore/grpc/aspnetcore.md [hunk 6]
```diff
@@ -246,6 +246,27 @@ For more information on enabling HTTP/2 and TLS with Kestrel, see [Kestrel endpo
...
+
+The preceding project file:
+
+* Adds a framework reference to `Microsoft.AspNetCore.App`. For more information, see [Use the ASP.NET Core shared framework](xref:fundamentals/target-aspnetcore#use-the-aspnet-core-shared-framework).
+  * The framework reference allows non-ASP.NET Core apps, such as Windows Services, WPF apps, or WinForms apps to use ASP.NET Core APIs.
+  * The app can now use ASP.NET Core APIs to start an ASP.NET Core server.
+* Adds gRPC requirements:
+  * NuGet package reference to [`Grpc.AspNetCore`](https://www.nuget.org/packages/Grpc.AspNetCore).
+  * `.proto` file.
+
```
Comment (JamesNK) [inline aspnetcore/grpc/aspnetcore.md:269]:
```suggestion
* Adds a framework reference to `Microsoft.AspNetCore.App`.
  * The framework reference allows non-ASP.NET Core apps, such as Windows Services, WPF apps, or WinForms apps to use ASP.NET Core APIs.
  * The app can now use ASP.NET Core APIs to start an ASP.NET Core server.
* Adds gRPC requirements:
  * NuGet package reference to [`Grpc.AspNetCore`](https://www.nuget.org/packages/Grpc.AspNetCore).
  * `.proto` file.

For more information about using the `Microsoft.AspNetCore.App` framework reference, see [Use the ASP.NET Core shared framework](xref:fundamentals/target-aspnetcore#use-the-aspnet-core-shared-framework).
```

**Pull request merged**